src/stan/io/empty_var_context.hpp

```diff
@@ -0,0 +1,113 @@
+#ifndef STAN_IO_EMPTY_VAR_CONTEXT_HPP
+#define STAN_IO_EMPTY_VAR_CONTEXT_HPP
+
+#include <stan/io/var_context.hpp>
+#include <string>
+#include <vector>
+
+namespace stan {
+  namespace io {
+
+    /**
+     * This is an implementation of a var_context that doesn't contain
+     * any variables.
+     */
+    class empty_var_context : public var_context {
+    public:
+      /**
+       * Destructor
+       *
+       */
+      ~empty_var_context() {}
+
+      /**
+       * Return <code>true</code> if the specified variable name is
+       * defined. Always returns <code>false</code>.
+       *
+       * @param name Name of variable.
+       * @return <code>false</code>
+       */
+      bool contains_r(const std::string& name) const {
+        return false;
+      }
+
+      /**
+       * Always returns an empty vector. 
+       *
+       * @param name Name of variable.
+       * @return empty vector
+       */
+      std::vector<double> vals_r(const std::string& name) const {
+        std::vector<double> vals_r;
+        return vals_r;
+      }
+
+      /**
+       * Always returns an empty vector. 
+       *
+       * @param name Name of variable.
+       * @return empty vector
+       */
+      std::vector<size_t> dims_r(const std::string& name) const {
+        std::vector<size_t> dims_r;
+        return dims_r;
+      }
+
+      /**
+       * Return <code>true</code> if the specified variable name has
+       * integer values. Always returns <code>false</code>.
+       *
+       * @param name Name of variable.
+       * @return false
+       */
+      bool contains_i(const std::string& name) const {
+        return false;
+      }
+
+      /**
+       * Returns an empty vector.
+       *
+       * @param name Name of variable.
+       * @return empty vector
+       */
+      std::vector<int> vals_i(const std::string& name) const {
+        std::vector<int> vals_i;
+        return vals_i;
+      }
+
+      /**
+       * Return the dimensions of the specified floating point variable.
+       * Returns an empty vector.
+       *
+       * @param name Name of variable.
+       * @return empty vector
+       */
+      std::vector<size_t> dims_i(const std::string& name) const {
+        std::vector<size_t> dims_i;
+        return dims_i;
+      }
+
+      /**
+       * Return a list of the names of the floating point variables in
+       * the context. This contet has no variables.
+       *
+       * @param names Vector to store the list of names in.
+       */
+      void names_r(std::vector<std::string>& names) const {
+        names.clear();
+      }
```

Comment:
This follows the design of the super class, `var_context`, which has been around for a long time.